### DISCUSSION TITLE
Usage of Vec in request

### DISCUSSION THREAD

**Discussion by averyoldman** [rwf2/Rocket#2572] [Questions] [answered]
Hi, 

I want to send an array of a certain struct to the interface.
When I define the method like this I get an error like:
the trait `FromData<'_>` is not implemented for `Vec<Rapport>`

I can't implement fromdata on vec cause fromdata is defined outside my package.
It also feels like adding alot of boilerplate code to wrap Rapport for example so I'd prefer not to do it.

Is there an easy way to use Vec here?

```
#[patch("/rapport",data="<rapport>")]
pub async fn update_rapport(conn: &State<Connection>, rapport: Vec<Rapport>) {
    println!("{:?}",rapport);
}
```

Kind regards,

An old man

**Comment by SergioBenitez** [comment, marked as answer]
Yes, but you need to specify the format. If it's [JSON](https://api.rocket.rs/v0.5-rc/rocket/serde/json/struct.Json.html), do:

```rust
#[patch("/rapport", data = "<rapport>")]
fn update_rapport(conn: &State<Connection>, rapport: Json<Vec<Rapport>>) { .. }
```

If it's a [form](https://api.rocket.rs/v0.5-rc/rocket/form/struct.Form.html), do:

```rust
#[patch("/rapport", data = "<rapport>")]
fn update_rapport(conn: &State<Connection>, rapport: Form<Vec<Rapport>>) { .. }
```

And so on.
